packages/core/src/ObservableActorRef.ts

```diff
@@ -46,9 +51,26 @@ export class ObservableActorRef<TEvent extends EventObject, TEmitted>
     };
   }
   public getSnapshot() {
-    return this.actor.current;
+    return this.current;
   }
   public [symbolObservable]() {
     return this;
   }
+  public receive(event: TEvent | LifecycleSignal) {
+    this.mailbox.enqueue(event);
+  }
+  private _process(event: TEvent | LifecycleSignal) {
+    this.context._event =
+      typeof event.type !== 'string'
+        ? (event as LifecycleSignal)
+        : toSCXMLEvent(event as TEvent);
+
+    this.current = this.behavior.transition(
```

Comment:
Let's not try to fix this here but I think there is something broken in how the pieces compose together here. It's unclear which piece is responsible for actually managing the snapshot's state, which piece is responsible for managing subscribers/observers, and also that there is a shared (& mutable) state in all of our existing behavior types (which has a nasty outcome - those behaviors are not truly reusable).

For example, if we remove the `this.current =` assignment from here then **only** 4 tests fail:
```
actors › with behaviors › should work with a reducer behavior
invoke › with behaviors › should work with a behavior
invoke › with reducers › should work with a reducer
invoke › with reducers › should schedule events in a FIFO queue
```

I have some thoughts related to this - let's maybe meet together and hash it out. I can do a followup refactor after we land this PR.

Reply:
@Andarist This is actually a really good catch; the reason this is happening is because two different things are trying to manage observers (the ObservableActorRef and the behaviors). This should only be done in the ObservableActorRef, so the behaviors become even smaller in scope (good thing).

I have a fix for this in progress.